hatchet_sdk/clients/dispatcher.py

```diff
@@ -170,10 +174,14 @@ def heartbeat(self):
                     timeout=5,
                     metadata=get_metadata(self.token),
                 )
+
+                self.last_heartbeat_succeeded = True
             except grpc.RpcError as e:
                 # we don't reraise the error here, as we don't want to stop the heartbeat thread
                 logger.error(f"Failed to send heartbeat: {e}")
 
+                self.last_heartbeat_succeeded = False
```

Comment:
does this catch all errors or just `except grpc.RpcError as e`?

Reply:
Just catches grpc errors - I think it's reasonable for other errors to bubble up